project-management/src/main/java/life/qbic/projectmanagement/domain/model/measurement/ProteomicsMeasurement.java

```diff
@@ -185,9 +193,8 @@ public void setSamplePreparation(ProteomicsSamplePreparation samplePreparation)
     this.comment = samplePreparation.comment();
   }
 
-  public void setLabeling(ProteomicsLabeling labeling) {
-    this.labelingType = labeling.labelType();
-    this.label = labeling.label();
+  public void setLabeling(Collection<ProteomicsLabeling> labeling) {
+    this.labeling = new HashSet<>(labeling.stream().toList());
```

Comment:
You can use `Set.copyOf(labeling)`. This creates an unmodifiable set that does not reflect any changes from the labeling collection.